frontend/src/store.ts

```diff
@@ -2649,6 +2649,17 @@ export function createStore(web3: Web3) {
           .call(defaultCallOptions(state));
       },
 
+      async isShieldPurchased({state}) {
+        const Promos = state.contracts().Promos!;
+        if (!state.defaultAccount || !Promos) return;
```

Comment:
Either use non-null assertion operator (**!**) on Promos or check if(!Promos), they both serve the same purpsoe here.